libs/hpc-live/src/lib/model.ts

```diff
@@ -394,6 +407,332 @@ export class LiveModel implements Model {
+          resultType: categories.GET_KEYWORDS_RESULT,
+        }),
+      deleteKeyword: (params) =>
+        this.call({
+          pathname: `/v2/category/${params.id}`,
+          method: 'DELETE',
+          resultType: categories.DELETE_KEYWORD_RESULT,
+        }),
+      updateKeyword: (params) =>
+        this.call({
+          pathname: `/v2/category/${params.id}`,
+          method: 'PUT',
+          body: {
+            type: 'json',
+            data: { data: { ...params } },
+          },
+          resultType: categories.CATEGORY,
+        }),
+    };
+  }
+  get emergencies(): emergencies.Model {
+    return {
+      getAutocompleteEmergencies: (params) =>
+        this.call({
+          pathname: `/v1/object/autocomplete/emergency/${params.query}`,
+          resultType: emergencies.GET_EMERGENCIES_AUTOCOMPLETE_RESULT,
+        }),
+    };
+  }
+  get systems(): systems.Model {
+    return {
+      getSystems: () =>
+        this.call({
+          pathname: '/v1/external/systems',
+          resultType: systems.GET_SYSTEMS_RESULT,
+        }),
+    };
+  }
+  get flows(): flows.Model {
+    return {
+      getFlowREST: (params) =>
+        this.call({
+          pathname: `/v2/flow/${params.id}`,
+          resultType: flows.GET_FLOW_RESULT,
+        }),
+      searchFlowsREST: (params) =>
+        this.call({
+          pathname: `/v2/flow/search`,
+          method: 'POST',
+          body: {
+            type: 'json',
+            data: params,
+          },
+          resultType: flows.SEARCH_FLOWS_RESULT_REST,
+        }),
+      getFlow: (params) =>
+        this.call({
+          pathname: `/v4/graphql`,
+          method: 'POST',
+          body: {
+            type: 'raw',
+            data: ` query Flow{
+              flow(id: ${params}) {
+                  createdAt
+                  updatedAt
+                  deletedAt
+                  id
+                  versionID
+                  amountUSD
+                  flowDate
+                  decisionDate
+                  firstReportedDate
+                  budgetYear
+                  origAmount
+                  origCurrency
+                  exchangeRate
+                  activeStatus
+                  newMoney
+                  restricted
+                  description
+                  notes
+                  versionStartDate
+                  versionEndDate
+                  createdBy
+                  lastUpdatedBy
+              }
+          }`,
+            contentType: 'application/json; charset=utf-8',
+          },
+          resultType: flows.GET_FLOW_RESULT,
+        }),
+      /**
+       * TODO: Dynamically fetch only necessary fields, Ex: if we don't display 'NewMoney' we shouldn't ask for it
+       */
+      searchFlows: (params) => {
+        const query = `query {
+          searchFlows${searchFlowsParams(params)} {
+            total
+            flows {
+              id
+              updatedAt
+              amountUSD
```

Comment:
I was looking into it and it could be implemented. The biggest advantage of using Apollo client (or any other), is the automatic management of the cache. In terms of code, the syntax is pretty similar actually, is just using a template string preceded by `gql`. If you want we could create a ticket to add Apollo client :)